src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilationModuleGroupBase.cs

```diff
@@ -363,8 +363,15 @@ public sealed override bool VersionsWithModule(ModuleDesc module)
 
         public sealed override bool VersionsWithType(TypeDesc typeDesc)
         {
-            return typeDesc.GetTypeDefinition() is EcmaType ecmaType &&
-                _versionsWithTypeCache.GetOrAdd(typeDesc, _versionsWithTypeUncached);
+            if (typeDesc.GetTypeDefinition() is EcmaType)
```

Comment:
It does not look right that we handle parametrized arrays here and generics in ComputeTypeVersionsWithCode. 

Should this method just "return _versionsWithTypeCache.GetOrAdd(typeDesc, _versionsWithTypeUncached)` and the handling of parametrized types to be all in ComputeTypeVersionsWithCode?